src/lib/src/receiver.cpp

```diff
@@ -110,6 +100,13 @@ int roc_receiver_bind(roc_receiver* receiver,
         return -1;
     }
 
+    if (receiver->addresses.size() == receiver->addresses.max_size()) {
+        if (!receiver->addresses.grow((receiver->addresses.size() + 1) * 2)) {
```

Comment:
I guess we can multiply the size by 2 each time we reach the `max_size` (no need to add 1). Something similar to:

```cpp
if (!receiver->addresses.grow(receiver->addresses.size() * 2)) { }
```

Reply:
> if (!receiver->addresses.grow(receiver->addresses.size() * 2)) { }

I agree that it would be nice to remove `+ 1`, but the suggested fix wont work when size() is zero. So we also need a check for the zero case.

I suggest the following growth sequence: 0, 2, 4, 8, 16, ...

I'd extract this into an Array method `bool grow_exp(size_t min_size)` ("exp" for "exponential"). It will check if the size is at least `min_size`. If it's not, it will grow the array according to the sequence.

Probably we could also reuse this method in core::StringList.

Reply:
@alexandremgo If you'll decide to extract it into Array method, please also add a corresponding test to `test_array.cpp`.

Reply:
Yes i did the `+1` because of the case when size() is 0. 

I am going to add the `bool grow_exp(size_t min_size)` method to Array

Reply:
If `bool grow_exp(size_t min_size)` is good, i can add the same method to core::StringList ;)

Reply:
LGTM!

> If bool grow_exp(size_t min_size) is good, i can add the same method to core::StringList ;)

Would be cool. This can be done in a separate PR I think.

Just to be sure: I didn't mean to add StringList::grow_exp(). I meant to re-use Array::grow_exp() in StringList::grow(), if it makes sense. (StringList uses Array under the hood).